src/games/behavmixed.imp

```diff
@@ -140,44 +109,28 @@ template <class T>
 MixedBehaviorProfile<T>::MixedBehaviorProfile(const MixedStrategyProfile<T> &p_profile)
   : m_probs(p_profile.GetGame()->NumActions()),
     m_support(p_profile.GetGame()),
-    m_gameversion(p_profile.GetGame()->GetVersion()),
-    m_cacheValid(false),
-    m_realizProbs(m_support.GetGame()->NumNodes()),
-    m_beliefs(m_support.GetGame()->NumNodes()),
-    m_nvals(m_support.GetGame()->NumNodes()),
-    m_bvals(m_support.GetGame()->NumNodes()),
-    m_nodeValues(m_support.GetGame()->NumNodes(),
-		 m_support.GetGame()->NumPlayers()),
-    m_infosetValues(m_support.GetGame()->NumInfosets()),
-    m_actionValues(m_support.GetGame()->NumActions()),
-    m_regret(m_support.GetGame()->NumActions())
-{
-  m_realizProbs = (T) 0.0;
-  m_beliefs = (T) 0.0;
-  m_nodeValues = (T) 0.0;
-  m_infosetValues = (T) 0.0;
-  m_actionValues = (T) 0.0;
-  m_regret = (T) 0.0;
-
+    m_gameversion(p_profile.GetGame()->GetVersion())
+{
   ((Vector<T> &) *this).operator=((T)0);
 
   GameTreeNodeRep *root = dynamic_cast<GameTreeNodeRep *>(m_support.GetGame()->GetRoot().operator->());
   const StrategySupportProfile &support = p_profile.GetSupport();
   GameRep *game = m_support.GetGame();
 
   for (auto player : game->GetPlayers()) {
-    m_nvals = (T) 0;
-    m_bvals = (T) 0;
-
+    // reset to 0 or clear of map_nvals and map_bvals needed
+    //for (auto &p: map_nvals) p.second = (T) 0;
+    //for (auto &p: map_bvals) p.second = (T) 0;
+    map_bvals.clear();
```

Comment:
Looking at this, `nvals` and `bvals` are scratch data, and so they really shouldn't be members of the class.  Instead it would seem to make more sense to have them be declared and initialised here in this function and then passed to the recursive calls.

They would have been made members of the class simply because in the old way of doing things allocation was expensive and so it seemed to make sense to have them be allocated once as scratch arrays and kept around.